src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -56,75 +56,97 @@ public CommandResult execute(Model model) {
         return new CommandResult(messageHeader + searchResult);
     }
 
+    /**
+     * Builds the message header for the command result.
+     *
+     * @param keyword The keyword used for filtering.
+     * @return The formatted message header.
+     */
     private String buildMessageHeader(String keyword) {
         return String.format(Messages.MESSAGE_SCHEDULES_LISTED, keyword) + "\n\n";
     }
 
+    /**
+     * Fetches the search result based on the given keyword.
+     *
+     * @param model The model containing the list of persons and schedules.
+     * @param keyword The keyword used for filtering.
+     * @return The formatted search result.
+     */
     private String fetchSearchResult(Model model, String keyword) {
+        LocalDate date = predicate.getDateToFind();
         return DayOfWeekUtils.isDayOfWeek(keyword)
-                ? resultGivenDay(model)
-                : resultGivenDate(model);
+                ? resultGiven(model, predicate.getDayToFind(), date)
+                : resultGiven(model, date.getDayOfWeek(), date);
```

Comment:
```suggestion
            resultGiven(model, predicate.getDayToFind(), predicate.getDateToFind());
```
Actually the code can be simplified further, since the day and date to find have been computed in the constructor for ScheduleContainsKeyword, so the isDayOfWeek conditional is not really necessary here. Only realised this after you helped to refactor the methods in the following lines. If possible can you help to simplify the code based on this logic?